includes/admin/import-functions.php

```diff
@@ -911,6 +918,64 @@ function give_save_import_donation_to_db( $raw_key, $row_data, $main_key = [], $
 				if ( 'pending' !== $status ) {
 					$payment->update_status( $status );
 				}
+
+                // Handle subscription import
+                if (
+                    defined('GIVE_RECURRING_VERSION')
+                    && ! empty( $data['payment_type'] ) && strtolower( $data['payment_type'] ) === 'subscription'
+                ) {
+                    global $wpdb;
+
+                    // Check if subscription already exists
+                    $subscriptionExists = DB::get_var(
+                        DB::prepare( "SELECT COUNT(id) FROM {$wpdb->prefix}give_subscriptions WHERE parent_payment_id = %d", $payment->ID )
```

Comment:
This logic will never get renewal donations associated with a subscription.  It can only identify if the first donation is from a subscription. As result, renewals will be imported without any association with the subscription.

<img width="1508" height="852" alt="image" src="https://github.com/user-attachments/assets/c7bd5b61-3eb6-4a36-8115-33b8ca5198b1" />

<img width="1507" height="855" alt="image" src="https://github.com/user-attachments/assets/a9934550-f2dc-4b58-9602-4ab2c78e4c69" />